front/src/modules/ui/input/relation-picker/hooks/useEntitySelectScroll.ts

```diff
@@ -4,24 +4,30 @@ import { Key } from 'ts-key-enum';
 import { useScopedHotkeys } from '@/ui/utilities/hotkey/hooks/useScopedHotkeys';
 import { useRecoilScopedState } from '@/ui/utilities/recoil-scope/hooks/useRecoilScopedState';
 
-import { relationPickerHoverIndexScopedState } from '../states/relationPickerHoverIndexScopedState';
-import { EntityForSelect } from '../types/EntityForSelect';
+import { CreateButtonId } from '../constants';
+import { RelationPickerRecoilScopeContext } from '../states/recoil-scope-contexts/RelationPickerRecoilScopeContext';
+import { relationPickerHoverIdScopedState } from '../states/relationPickerHoverIdScopedState';
 import { RelationPickerHotkeyScope } from '../types/RelationPickerHotkeyScope';
 
-export const useEntitySelectScroll = <
-  CustomEntityForSelect extends EntityForSelect,
->({
+export const useEntitySelectScroll = ({
   containerRef,
-  entities,
+  hoverIds,
 }: {
-  entities: CustomEntityForSelect[];
+  hoverIds: string[];
   containerRef: React.RefObject<HTMLDivElement>;
 }) => {
-  const [relationPickerHoverIndex, setRelationPickerHoverIndex] =
-    useRecoilScopedState(relationPickerHoverIndexScopedState);
+  const [relationPickerHoverId, setRelationPickerHoverId] =
+    useRecoilScopedState(
+      relationPickerHoverIdScopedState,
+      RelationPickerRecoilScopeContext,
+    );
+  const getIndex = hoverIds.findIndex((val) => val === relationPickerHoverId);
```

Comment:
You can put this in a selector, see : https://recoiljs.org/docs/api-reference/utils/selectorFamily And you have an example here : `numberOfTableColumnsScopedSelector`

Reply:
If it's too complicated with a selector, try to create a util function at least to abstract this away.